plugins/plugin-factory-github/codenvy-plugin-factory-github-server/src/main/java/com/codenvy/plugin/factory/github/server/GithubFactoryParametersResolver.java

```diff
@@ -0,0 +1,194 @@
+import java.util.regex.Pattern;
+
+import static java.util.Collections.singletonList;
+import static org.eclipse.che.dto.server.DtoFactory.newDto;
+
+/**
+ * Provides Factory Parameters resolver for github repositories.
+ *
+ * @author Florent Benoit
+ */
+public class GithubFactoryParametersResolver implements FactoryParametersResolver {
+
+    /**
+     * Logger.
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(GithubFactoryParametersResolver.class);
+
+    /**
+     * Parameter name.
+     */
+    protected static final String URL_PARAMETER_NAME = "url";
+
+    /**
+     * Connection timeout of 30seconds.
+     */
+    private static final int CONNECTION_TIMEOUT = (int) TimeUnit.SECONDS.toMillis(30);
+
+    /**
+     * Regexp to find repository details (repository name, project name and branch and subfolder)
+     */
+    private final Pattern GITHUB_PATTERN = Pattern.compile("^(?:http)(?:s)?(?:\\:\\/\\/)github.com/(.*?)/(.*?)(?:/tree/(.*?)(?:/(.*?))?)?$");
+
+    /**
+     * Check if this resolver can be used with the given parameters.
+     *
+     * @param factoryParameters
+     *         map of parameters dedicated to factories
+     * @return true if it will be accepted by the resolver implementation or false if it is not accepted
+     */
+    @Override
+    public boolean accept(@NotNull Map<String, String> factoryParameters) {
+        // Check if url parameter is a github URL
+        return factoryParameters.containsKey(URL_PARAMETER_NAME) && GITHUB_PATTERN.matcher(factoryParameters.get(URL_PARAMETER_NAME)).matches();
+    }
+
+    /**
+     * Create factory object based on provided parameters
+     *
+     * @param factoryParameters
+     *         map containing factory data parameters provided through URL
+     * @throws BadRequestException when data are invalid
+     */
+    @Override
+    public Factory createFactory(@NotNull Map<String, String> factoryParameters) throws BadRequestException {
+
+        String githubUrl = factoryParameters.get("url");
+
+        // Apply github url to the regexp
+        Matcher matcher = GITHUB_PATTERN.matcher(githubUrl);
+        if (!matcher.matches()) {
+            throw new BadRequestException(String.format(
+                    "The given github url %s is not a valid URL github url. It should start with https://github.com/<user>/<repo>",
+                    githubUrl));
+        }
+
+        // ok so now get matching group
+        int regexpGroupCount = matcher.groupCount();
+        if (regexpGroupCount < 2) {
+            throw new BadRequestException(String.format(
+                    "The given github url %s is not a valid URL github url. It should start with https://github.com/<user>/<repo>",
+                    githubUrl));
+        }
+
+        String repoUser = matcher.group(1);
+        String repoName = matcher.group(2);
+        String branchName = null;
+        String subfolder = null;
+        if (regexpGroupCount == 4) {
```

Comment:
`https://github.com/eclipse/che/tree/master` fits regexp and has 3 groups. Don't you think it is valid usecase?